lightblue/src/main/java/org/esbtools/eventhandler/lightblue/client/BulkLightblueRequester.java

```diff
@@ -349,20 +353,59 @@ public boolean isCancelled() {
         public boolean isDone() {
             return cancelled || completed;
         }
+        
+        private static class TimeoutDuration {
+            final long duration;
+            final TimeUnit timeUnit;
+
+            TimeoutDuration(long duration, TimeUnit timeUnit) {
+
+                this.duration = duration;
+                this.timeUnit = timeUnit;
+            }
+
+        }
 
         @Override
         public U get() throws InterruptedException, ExecutionException {
+            try {
+                return get(Optional.empty());
+            } catch (TimeoutException e) {
+                throw new ExecutionException("A future without a timeout is attempting to throw a TimeoutException while it should not. "
+                                + "This implies there is a bug in the future code and will need to be corrected ",e);
+            }
+        }
+
+        @Override
+        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+
+            return get(Optional.of(new TimeoutDuration(timeout, unit)));
+        }
+
+        private U get(Optional<TimeoutDuration> timeout)
+                throws TimeoutException, InterruptedException, ExecutionException {
             if (cancelled) {
                 throw new CancellationException();
             }
 
             if (!completed) {
-                completer.triggerFutureCompletion();
-
+                if (timeout.isPresent()) {
+                    ExecutorService executor = Executors.newSingleThreadExecutor();
+                    try {
+                        Future<?> submittedTask = executor.submit(() -> {
+                            completer.triggerFutureCompletion();
+                        });
+                        submittedTask.get(timeout.get().duration, timeout.get().timeUnit);
```

Comment:
semi-nitpicky: I'd consider doing one `TimeoutDuration duration = timeout.get();` but not a big deal